### PR TITLE
Roll forward Dataflow V2 portable job submission changes from #14074 but without making it the default

### PR DESCRIPTION
**Pull request by kennknowles**
This should make it easy to test whether making it the default is safe, by rolling forward all the sickbay lists and fixes that were necessary to get it green.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Java Examples on Dataflow Runner V2

**Comment by boyuanzz** [comment]
Hi Kenn,

Any reason that we want to separate PR14297 and this one? It would be easy to rollback or check the history if all changes are in the same PR.

**Comment by kennknowles** [comment]
Only because this seemed unhealthy that that change seemed the minimum to make progress.

**Comment by kennknowles** [comment]
My prefernce is to make steady progress that will not need to be rolled back. Multiple commits in the same PR are fine as long as whoever is rolling back knows how to bisect.

**Comment by kennknowles** [comment]
More specifically

 - Ideally the sickbay change is kept as that is the true sickbay for non-deprecated runner V2 paths.
 - The portable pipeline being uploaded today is somewhat wrong for runner v2. Making that right is somewhat separable from the remaining changes.

But if this PR can go in and be green that is even better.

**Comment by kennknowles** [comment]
run java postcommit

**Comment by kennknowles** [comment]
Run Java Examples on Dataflow Runner V2

**Comment by kennknowles** [comment]
If Pubsub works and examples work, I am confident in our understanding of ValidatesRunner results. No need to waste resources running that eagerly.

**Comment by kennknowles** [comment]
Well, nvm of course this still should be turning on portable job submission so we expect everything to be unchanged.

**Comment by kennknowles** [comment]
Merging in changes to apply portable replacements.

**Comment by kennknowles** [comment]
run java postcommit

**Comment by kennknowles** [comment]
Run Java Examples on Dataflow Runner V2

**Review by boyuanzz — APPROVED** [review]
Thanks for making these changes.

**Comment by kennknowles** [comment]
Every single pipeline failed with the same cause: `The worker lost contact with the service.`

No stages of any pipeline succeeded.

**Comment by kennknowles** [comment]
Run Java Examples on Dataflow Runner V2

**Comment by kennknowles** [comment]
I pulled job ids from https://ci-beam.apache.org/job/beam_PostCommit_Java_Examples_Dataflow_V2_PR/18/ and quickly browsed the logs.

If this is not an environmental problem, then there is probably something obviously wrong with the change somehow. Maybe it breaks non-portable job submission?

**Comment by kennknowles** [comment]
If course, it would break Pubsub.

**Comment by kennknowles** [comment]
Actualy no.

**Comment by boyuanzz** [comment]
Checking job: 2021-03-24_17_34_31-11200910305787742169

It seems like certain step is missing environment id: https://screenshot.googleplex.com/6VDZsbds8DWfgS2.png

**Comment by kennknowles** [comment]
Ah! Comparing the log view that pops open when I hit the button, I see the "Go To Logs Explorer" button is context-sensitive so I was clicking it in a context that filtered for only job messages.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java [hunk 1]
```diff
@@ -1177,7 +1232,7 @@ public DataflowPipelineJob run(Pipeline pipeline) {
             jobResult.getId(),
             options,
             jobSpecification.getStepNames(),
-            pipelineProto);
+            dataflowV1PipelineProto);
```
Comment (kennknowles) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java:1235]: This one is supposed to be a V2 proto.

**Comment by kennknowles** [comment]
Run Java Examples on Dataflow Runner V2

**Comment by kennknowles** [comment]
Run Java Examples on Dataflow Runner V2

**Comment by kennknowles** [comment]
Oh I think I figured it out and it is a bit obvious: the staged pipeline is portable so the steps in the v1beta3 graph do not exist. So it just makes sense to roll it all forward together. I opened #14339 and will test it for the regressions that happened last time.

**Comment by kennknowles** [comment]
The alternative is to adjust this PR to stage the V1 proto, and the follow-up that turns it on will both set the flag and change with proto is uploaded.

**Comment by kennknowles** [comment]
Merged #14339

**Pull request closed without merging**